src/silx/gui/plot/backends/glutils/GLText.py

```diff
@@ -267,21 +318,32 @@ def render(self, matrix: numpy.ndarray):
         texUnit = 0
         texture, offset = self._getTexture()
 
-        gl.glUniform1i(prog.uniforms["texText"], texUnit)
-
         mat = numpy.dot(matrix, mat4Translate(int(self.x), int(self.y)))
+
+        vertices = self.getVertices(offset, texture.shape)
+        bgVertices = self.getBgVertices(offset, texture.shape)
+
         gl.glUniformMatrix4fv(
             prog.uniforms["matrix"], 1, gl.GL_TRUE, mat.astype(numpy.float32)
         )
-
-        gl.glUniform4f(prog.uniforms["color"], *self.color)
         if self.bgColor is not None:
             bgColor = self.bgColor
         else:
             bgColor = self.color[0], self.color[1], self.color[2], 0.0
         gl.glUniform4f(prog.uniforms["bgColor"], *bgColor)
+        posAttrib = prog.attributes["position"]
+        gl.glEnableVertexAttribArray(posAttrib)
+        gl.glVertexAttribPointer(posAttrib, 2, gl.GL_FLOAT, gl.GL_FALSE, 0, bgVertices)
+        gl.glDrawArrays(gl.GL_TRIANGLE_STRIP, 0, 4)
 
-        vertices = self.getVertices(offset, texture.shape)
+        gl.glUniform1i(prog.uniforms["texText"], texUnit)
```

Comment:
That's rendering without setting the texture, you might get something that was set before. I'll check it.

Reply:
Yes, you have to render twice, 1 for the background, 1 for the text

Because the boxes are not the same. But it can be wrong at some place.